src/test/java/com/github/shyiko/mysql/binlog/event/deserialization/BinaryLogEventDataReaderTest.java

```diff
@@ -0,0 +1,478 @@
+        byte[] data = {(byte)0xfe, (byte)0xa2, 0x12, 0x45, 0x11, (byte)0xd1, 0x11, 0x22, 0x33, 0x44};
+        BinaryLogEventDataReader reader = new BinaryLogEventDataReader(ByteBuffer.wrap(data));
+
+        reader.readPackedInteger();
+    }
+
+    @Test
+    public void readString_shouldProperlyDecodeStrings() {
+        byte[] data = {0x66, 0x6f, 0x6f, 0x2d, 0x62, 0x61, 0x72, 0x12, 0x61};
+        BinaryLogEventDataReader reader = new BinaryLogEventDataReader(ByteBuffer.wrap(data));
+
+        String value = reader.readString(7);
+
+        assertEquals("foo-bar", value);
+        assertEquals(2, reader.available());
+    }
+
+    @Test
+    public void readLengthEncodedString_shouldProperlyDecodeStrings() throws IOException {
+        byte[] data = {0x07, 0x66, 0x6f, 0x6f, 0x2d, 0x62, 0x61, 0x72, 0x12, 0x61};
+        BinaryLogEventDataReader reader = new BinaryLogEventDataReader(ByteBuffer.wrap(data));
+
+        String value = reader.readLengthEncodedString();
+
+        assertEquals("foo-bar", value);
+        assertEquals(2, reader.available());
+    }
+
+    @Test
+    public void readZeroTerminatedStringWithHint_shouldProperlyDecodeStrings() throws IOException {
+        byte[] data = {0x66, 0x6f, 0x6f, 0x2d, 0x62, 0x61, 0x72, 0x00, 0x12};
+        BinaryLogEventDataReader reader = new BinaryLogEventDataReader(ByteBuffer.wrap(data));
+
+        String value = reader.readZeroTerminatedStringWithHint(7);
+
+        assertEquals("foo-bar", value);
+        assertEquals(1, reader.available());
+    }
+
+    @Test
+    public void readZeroTerminatedString_shouldProperlyDecodeStrings() throws IOException {
+        byte[] data = {0x66, 0x6f, 0x6f, 0x2d, 0x62, 0x61, 0x72, 0x00, 0x12};
+        BinaryLogEventDataReader reader = new BinaryLogEventDataReader(ByteBuffer.wrap(data));
+
+        String value = reader.readZeroTerminatedString();
+
+        assertEquals("foo-bar", value);
+        assertEquals(1, reader.available());
+    }
+
+    @Test(expected = IOException.class)
+    public void readZeroTerminatedString_shouldThrowIOExceptionWithNoZeroByte() throws IOException {
+        byte[] data = {0x66, 0x6f, 0x6f, 0x2d, 0x62, 0x61, 0x72, 0x12};
+        BinaryLogEventDataReader reader = new BinaryLogEventDataReader(ByteBuffer.wrap(data));
+
+        reader.readZeroTerminatedString();
+    }
+
+    @Test(expected = IOException.class)
+    public void readZeroTerminatedStringWithHint_shouldThrowExceptionIfStringIsNotZeroTerminated() throws IOException {
+        byte[] data = {0x66, 0x6f, 0x6f, 0x2d, 0x62, 0x61, 0x72, 0x12};
+        BinaryLogEventDataReader reader = new BinaryLogEventDataReader(ByteBuffer.wrap(data));
+
+        reader.readZeroTerminatedStringWithHint(7);
+    }
+
+    @Test
+    public void readBitSet_shouldProperlyDecodeLittleEndianBitSets() {
+        BitSetTestData[] testData = {
+            // 0101 1101 0001 0001 -> read 1 byte
+            // 0101 1101 -> take 8 bits
```

Comment:
I think this comment should say 4 bits?
```suggestion
            // 0101 1101 -> take 4 bits
```

Reply:
Fixed.